index.d.ts

```diff
@@ -18,64 +26,22 @@ import {
   SecureClientSessionOptions,
 } from "http2";
 
-declare function fastifyReplyFrom<HttpServer, HttpRequest, HttpResponse>(
-  instance: fastify.FastifyInstance<HttpServer, HttpRequest, HttpResponse>,
-  opts: fastifyReplyFrom.ReplyFromOptions<
-    HttpServer,
-    HttpRequest,
-    HttpResponse
-  >,
-  callback?: (err?: Error) => void
-): void;
-
-interface Http2Options {
-  sessionTimeout?: number;
-  requestTimeout?: number;
-  sessionOptions?: ClientSessionOptions | SecureClientSessionOptions;
-  requestOptions?: ClientSessionRequestOptions;
-}
-
-interface HttpOptions {
-  agentOptions?: AgentOptions | SecureAgentOptions;
-  requestOptions?: RequestOptions | SecureRequestOptions;
-}
-
-declare namespace fastifyReplyFrom {
-  interface ReplyFromOptions<HttpServer, HttpRequest, HttpResponse>
-    extends fastify.RegisterOptions<HttpServer, HttpRequest, HttpResponse> {
-    base?: string;
-    cacheURLs?: number;
-    http?: HttpOptions;
-    http2?: Http2Options | boolean;
-    undici?: unknown; // undici has no TS declarations yet
-    keepAliveMsecs?: number;
-    maxFreeSockets?: number;
-    maxSockets?: number;
-    rejectUnauthorized?: boolean;
-    sessionTimeout?: number;
-  }
-}
-
-export = fastifyReplyFrom;
-
 declare module "fastify" {
-  interface FastifyReply<HttpResponse> {
+  interface FastifyReplyInterface {
     from(
       source?: string,
       opts?: {
         queryString?: { [key: string]: unknown };
         contentType?: string;
         onResponse?: (
-          request: FastifyRequest,
-          reply: FastifyReply<HttpResponse>,
-          res: unknown
+          request: FastifyRequest<RawServerBase>,
```

Comment:
`RawServerBase` is `http.Server | https.Server | http2.Http2Server | http2.Http2SecureServer`
`RawServerDefault` is just `http.Server`

We do want `RawServerBase` here so that the correct union type fro raw request is inferred: `http.IncomingMessage | http2.Http2ServerRequest`

Reply:
Ahh I see. Okay lgtm